src/ocean/text/convert/Formatter.d

```diff
@@ -962,3 +977,163 @@ unittest
     static assert(canSwitchOn!(immutable int));
     static assert(!canSwitchOn!(real));
 }
+
+/// Returns `true` if `R` is an input range.
```

Comment:
All of this was extracted from Phobos, right ?

Reply:
I referred to Phobos. I made a function that it does not support. And some of the test codes will be similar.

Reply:
I added a link to the reference code below.
```
    See_Also:
        https://github.com/dlang/phobos/blob/
        418106123d6106efb03b2753bf191215b411ad96/std/range/primitives.d#L124
```